tests/kubernetes/minikube.py

```diff
@@ -0,0 +1,204 @@
+            return self.client
+        else:
+            return None
+
+    def get_ip(self):
+        if self.container:
+            self.ip = self.container.attrs["NetworkSettings"]["IPAddress"]
+            return self.ip
+        else:
+            return None
+
+    def load_kubeconfig(self, kubeconfig_path="/kubeconfig", timeout=300):
+        assert wait_for(p(container_cmd_exit_0, self.container, "test -f %s" % kubeconfig_path), timeout_seconds=timeout), "timed out waiting for the minikube cluster to be ready!\n\n%s\n\n" % self.container.logs().decode('utf-8').strip()
+        self.kubeconfig = "/tmp/scratch/kubeconfig-%s" % self.container.id[:12]
+        time.sleep(2)
+        rc, output = self.container.exec_run("cp -f %s %s" % (kubeconfig_path, self.kubeconfig))
+        assert rc == 0, "failed to get %s from minikube!\n%s" % (kubeconfig_path, output.decode('utf-8'))
+        time.sleep(2)
+        kube_config.load_kube_config(config_file=self.kubeconfig)
+
+    def connect(self, name, timeout, version=None):
+        print("\nConnecting to %s container ..." % name)
+        self.container = self.host_client.containers.get(name)
+        self.client = self.get_client()
+        self.name = name
+        self.version = version
+        self.load_kubeconfig(timeout=timeout)
+
+    def deploy(self, version, timeout, name=None, options={}):
+        self.version = version
+        if self.version[0] != 'v':
+            self.version = 'v' + self.version
+        self.name = name
+        if not self.name:
+            self.name = "minikube-%s-%s" % (MINIKUBE_VERSION, self.version)
+        if not options:
+            options = {
+                "name": self.name,
+                "privileged": True,
+                "environment": {
+                    'K8S_VERSION': self.version,
+                    'TIMEOUT': str(timeout)
+                },
+                "ports": {
+                    '8080/tcp': None,
+                    '8443/tcp': None,
+                    '2375/tcp': None,
+                },
+                "volumes": {
+                    "/tmp/scratch": {
+                        "bind": "/tmp/scratch",
+                        "mode": "rw"
+                    },
+                }
+            }
+        print("\nDeploying minikube %s cluster ..." % self.version)
+        image, logs = self.host_client.images.build(
+            path=os.path.join(TEST_SERVICES_DIR, 'minikube'),
+            buildargs={"MINIKUBE_VERSION": MINIKUBE_VERSION},
+            tag="minikube:%s" % MINIKUBE_VERSION,
+            rm=True,
+            forcerm=True)
+        self.container = self.host_client.containers.run(
+            image.id,
+            detach=True,
+            **options)
+        self.load_kubeconfig(timeout=timeout)
+        self.container.reload()
+        self.get_client()
+        self.get_ip()
```

Comment:
What is the point of calling this?  It doesn't appear to cause any side effects and isn't used in any way.